src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -33,30 +35,38 @@ object ClusterMonitorSupervisor {
   case class ClusterStarted(cluster: Cluster) extends ClusterSupervisorMessage
   // sent after a cluster is updated by the user
   case class ClusterUpdated(cluster: Cluster) extends ClusterSupervisorMessage
-
   // sent after cluster creation fails, and the cluster should be recreated
   case class RecreateCluster(cluster: Cluster) extends ClusterSupervisorMessage
   // sent after cluster creation succeeds, and the cluster should be stopped
   case class StopClusterAfterCreation(cluster: Cluster) extends ClusterSupervisorMessage
+  //Sent when the cluster should be removed from the monitored cluster list
+  case class RemoveFromList(cluster: Cluster) extends ClusterSupervisorMessage
   // Auto freeze idle clusters
   case object AutoFreezeClusters extends ClusterSupervisorMessage
+  // Timers ADT
+  sealed trait TimerTick extends ClusterSupervisorMessage
+  private case object TimerKey extends TimerTick
+  private case object Tick extends TimerTick
 }
 
-class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: DataprocConfig, gdDAO: GoogleDataprocDAO, googleComputeDAO: GoogleComputeDAO, googleIamDAO: GoogleIamDAO, googleStorageDAO: GoogleStorageDAO, dbRef: DbReference, authProvider: LeoAuthProvider, autoFreezeConfig: AutoFreezeConfig, jupyterProxyDAO: JupyterDAO) extends Actor with LazyLogging {
+class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: DataprocConfig, gdDAO: GoogleDataprocDAO, googleComputeDAO: GoogleComputeDAO, googleIamDAO: GoogleIamDAO, googleStorageDAO: GoogleStorageDAO, dbRef: DbReference, authProvider: LeoAuthProvider, autoFreezeConfig: AutoFreezeConfig, jupyterProxyDAO: JupyterDAO, leonardoService: LeonardoService)
+  extends Actor with Timers with LazyLogging {
   import context.dispatcher
 
-  var leoService: LeonardoService = _
+  val monitoredClusters: ValueBox[Set[Cluster]] = ValueBox(Set.empty)
 
   import context._
 
   override def preStart(): Unit = {
     super.preStart()
-    if(autoFreezeConfig.enableAutoFreeze)
+
+    timers.startPeriodicTimer(TimerKey, Tick, monitorConfig.pollPeriod)
+
+    if (autoFreezeConfig.enableAutoFreeze)
       system.scheduler.schedule(autoFreezeConfig.autoFreezeCheckScheduler, autoFreezeConfig.autoFreezeCheckScheduler, self, AutoFreezeClusters)
```

Comment:
I see akka `Timer` and `Scheduler` being used here -- can we be consistent re timer vs. scheduler?